src/hedgehog.js

```diff
@@ -62,6 +62,42 @@ class Hedgehog {
     }
   }
 
+  /**
+   * Generate new set of auth credentials to allow login
+   * TODO: Expose way to eliminate previous credentials
+   * @param {String} username - username
+   * @param {String} password - new password
+   * @param {String} entropy - stored entropy val
+   */
+  async resetPassword (username, password) {
+    let self = this
+    let entropy = await WalletManager.getEntropyFromLocalStorage()
+    if (entropy === null) {
+      throw new Error('resetPassword - missing entropy')
+    }
+
+    const createWalletPromise = WalletManager.createWalletObj(password, entropy)
+    const lookupKeyPromise = WalletManager.createAuthLookupKey(username, password)
+    try {
+      let result = await Promise.all([createWalletPromise, lookupKeyPromise])
+
+      const { ivHex, cipherTextHex, walletObj } = result[0]
+      const lookupKey = result[1]
+
+      const authData = {
+        iv: ivHex,
+        cipherText: cipherTextHex,
+        lookupKey: lookupKey
+      }
+
+      console.log('Resetting auth fn...')
+      await self.setAuthFn(authData)
```

Comment:
@hareeshnagaraj just add a todo here to delete the old auth record, but i think for now this is a decent start

Reply:
added at top of function